src/OpenTelemetry.Api/Context/RuntimeContext.cs

```diff
@@ -27,10 +28,66 @@ public static class RuntimeContext
     {
         private static readonly ConcurrentDictionary<string, object> Slots = new();
 
+        private static Type contextSlotType;
+
+        private static RuntimeContextSlotFactory runtimeContextSlotFactory;
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode", Justification = "User-defined RuntimeContextSlotFactory that relies on Reflection is not timmer safe.")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL3050:RequiresDynamicCode", Justification = "User-defined RuntimeContextSlotFactory that relies on Reflection is not AoT safe.")]
+        static RuntimeContext()
+        {
+            ContextSlotType = typeof(AsyncLocalRuntimeContextSlot<>);
+        }
+
         /// <summary>
         /// Gets or sets the actual context carrier implementation.
         /// </summary>
-        public static Type ContextSlotType { get; set; } = typeof(AsyncLocalRuntimeContextSlot<>);
+        public static Type ContextSlotType
+        {
+            get
+            {
+                return contextSlotType;
+            }
+
+            [RequiresUnreferencedCode("ReflectionRuntimeContextSlotFactory is trimmer unsafe.")]
+            [RequiresDynamicCode("ReflectionRuntimeContextSlotFactory requires the ability to generate new code at runtime.")]
```

Comment:
These messages show up to users as warnings, so they need to read and understand them to know why the API is not compatible. `ReflectionRuntimeContextSlotFactory` is not a public type, so callers don't know what it is.

```suggestion
            [RequiresUnreferencedCode("Setting a custom ContextSlotType requires using Reflection to initialize the context slot.")]
            [RequiresDynamicCode("Setting a custom ContextSlotType requires using MakeGenericType to initialize the context slot. The native code for the generic type might not be available at runtime.")]
```